File: 04-eval-injection/vuln_code.js
'use strict';

/**
 * Exercise 04 - eval() Injection
 * WORKSHOP: Secure Programming
 *
 * Audit: Read this code carefully. Can you spot the security flaw?
 *
 * A simple CLI calculator: reads a math expression and prints the result.
 */

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

rl.question('Enter a math expression (e.g. 2+2): ', (input) => {
    try {
        // BUG: eval() executes ANY JavaScript, not just math!
        const result = eval(input);
        console.log(`Result: ${result}`);
    } catch (err) {
        console.error(`Error: ${err.message}`);
    }
    rl.close();
});

File: 01-tutorial-command-injection/vuln_code.py
#!/usr/bin/env python3

"""
Tutorial 01 - Command Injection
WORKSHOP: Secure Programming

Audit: Read this code carefully. Can you spot the security flaw?
"""

import os
import subprocess

def get_user(user):
    """Get informataion about a user."""
    result = subprocess.run(
          ["id", user],
          capture_output=True,
          text=True,
    )

if __name__ == "__main__":
    user = input("Enter username: ")
    get_user(user)
